packaging/observability/render-templates.sh

```
@@ -0,0 +1,91 @@
+
+    # Step 2: Render all
+    for pair in "${RENDER_MAP[@]}"; do
+        IFS='|' read -r template_file output_file <<< "$pair"
+        full_template_path="${templates_dir}/${template_file}"
+
+        # Special handling for UserInfo proxy - skip if not configured
+        if [[ "$template_file" == "flightctl-userinfo-proxy.container.template" ]]; then
+            if [[ -z "${USERINFO_UPSTREAM_URL:-}" ]]; then
+                log "INFO" "Skipping $template_file (USERINFO_UPSTREAM_URL not configured)"
+                # Remove the service from the restart list
+                unset SYSTEMD_SERVICES["flightctl-userinfo-proxy.service"]
```

Shellcheck (0.10.0)

[warning] Quote arguments to unset so they're not glob expanded.

(SC2184)